ganga/GangaCore/Core/GangaRepository/DStreamer.py

```diff
@@ -0,0 +1,529 @@
+        json_content["master"] = master
+
+    if json_content["type"] == "Job":
+        # `id` is used for indexing by mongo.
+        # json_content["id"] = json_content["id"]
+        result = document.replace_one(
+            filter={"id": json_content["id"], "master": json_content["master"]}, replacement=json_content, upsert=True,
+        )
+    else:
+        result = document.insert_one(json_content)
+
+    if result is None:
+        logger.debug(f"to_database error for object {j}")
+        raise DatabaseError(
+            Exception,
+            f"Object could not be inserted in the document linked by {document.name}. Insertion resulted in: {result}",
+        )
+    return result
+
+
+def object_from_database(_filter, document):
+    """Load JobObject from a json filestream
+
+    Arguments:
+    _filter (dict): The key-value pair used to search the object in the document
+    document: The document in the database where the object is stored
+    """
+    logger.debug("object_from_database")
+    content = document.find_one(filter=_filter)
+    if content is None:
+        logger.debug(
+            f"to_database error for `filter` {_filter} and `document` {document.name}"
+        )
+        raise DatabaseError(
+            Exception,
+            f"{_filter} pair was not found in the document linked by {document.name}",
+        )
+    loader = JsonLoader()
+    obj, error = loader.parse_static(content)
+    return obj, error
+
+
+def index_to_database(data, document):
+    """Save the index information into the `index` document of the database
+
+    Args:
+        data : To be added
+        document : To be added
+    """
+    logger.debug("index_to_database")
+    if data:
+        data["modified_time"] = time.time()
+        if "id" in data and "master" in data:
+            result = document.replace_one(
+                filter={"id": data["id"], "master": data["master"]}, replacement=data, upsert=True,
+            )
+        else:
+            result = document.insert_one(data)
+
+        if result is None:
+            raise DatabaseError(
+                Exception,
+                f"index could not be inserted in the document linked by {document.name}. Insertion resulted in: {result}",
+            )
+
+        return result
+
+
+def index_from_database(_filter, document, many=False):
+    """Save the index information into the `index` document of the database
+
+    Args:
+        _filter : To be added
+        document : To be added
+    """
+    logger.debug("index_from_database")
+    if many:
+        result = [*document.find(filter=_filter)]
+    else:
+        result = document.find_one(filter=_filter)
+
+    # if result is None:
+    #     raise DatabaseError(
+    #         Exception,
+    #         f"index could not be extracted in the document linked by {document.name}. Extracted resulted in: {result}:{_filter}",
+    #     )
+    return result
+
+
+# TODO: Remove the versioning
```

Comment:
We should remember about this schema issue. Maybe I will just add it as a new issue in GitHub.